File: 214/projects/03/java/Average.java
/* Average.java implements a grade average function using a switch
 * statement
 *
 * CS 214 Proj 03 by Duncan Van Keulen
 * On 2-20-2020
 ****************************************************************/

import java.util.Scanner;

public class Average {

   public static void main(String[] args) {
      Scanner keyboard = new Scanner(System.in);
      System.out.println("\nEnter your average: "); // prompt for avg

      // Make sure the user enters an integer
      try {
         int average = Integer.parseInt(keyboard.next()); // read avg
         System.out.println(letterGrade(average)); // display letter grade
      }
      catch (Exception e) {
         System.out.println("Error, you must enter an integer");
         main(null);
      }
      
   }
   /***************************************************************
    * yearCode() converts an academic year into its integer code.
    *
    * Receive: average, an int
    * PRE: average (input) is an int
    * Return: the letter grade corresponding to the entered average
    */

   public static char letterGrade(int average) {
      switch (average / 10) {
      case 10:
      case 9:
         return 'A';
      case 8:
         return 'B';
      case 7:
         return 'C';
      case 6:
         return 'D';
      default:
         return 'F';
      }
   }
}

File: 214/projects/03/clojure/src/average.clj
;;;; average.clj is a driver for function averageGrade.
;;;;
;;;; Input: An academic average score
;;;; Output: The letter grade for that academic average
;;;;         (100, 80, 50... for valid entries; 
;;;;
;;;; Usage: clojure -m average
;;;;
;;;; Begun by: Prof. Adams, for CS 214 at Calvin College.
;;;; Completed by: Duncan Van Keulen
;;;; Date: 2-20-2020
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns average)                    ; name the program

    ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
    ;;; average() returns the code for a given academic average.
    ;;; Receive: average, an integer.
    ;;; Precondition: average is an integer
    ;;; Return: the letter grade corresponding to the average.
    ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

    (defn average [avg]
       (= avg (/ avg 10))
       (cond
         (>= avg 100)  "A"
         (>= avg 90)   "A"
         (>= avg 80)   "B"
         (>= avg 70)   "C"
         (>= avg 60)   "D"
         :else       "F"
       )
    )
    
    ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
    ;;; Function -main() test-drives our average() function.
    ;;; Input: average: an integer
    ;;; Output: the letter grade corresponding to average.
    ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
    (defn -main []
     (print "\nEnter your academic average: ") (flush)
     (let
        [ avg (read) ]                                           ; read average (an integer)
        (println " Your letter grade: " (average avg) "\n")      ; display code of average's output
     )
    )
    
    